### PR TITLE
Fix profile backgrounds

### PR DESCRIPTION
**Pull request by tsunyoku**
uses s3 to upload & fetch profile backgrounds

### CODE REVIEW

File: app/handlers/profiles/settings/profbackground.go [hunk 1]
```diff
@@ -99,22 +128,45 @@ func ProfileBackgroundSubmitHandler(c *gin.Context) {
...
+			// TODO: CacheControl?
+		})
+		if err != nil {
+			m = msg.ErrorMessage{lu.T(c, "We were not able to save your profile background.")}
+			c.Error(err)
+			slog.ErrorContext(c, err.Error())
+			return
+		}
+
+		saveProfileBackground(ctx, 3, fmt.Sprintf("%d.gif", ctx.User.ID))
```
Comment (cmyui) [inline app/handlers/profiles/settings/profbackground.go:175]: what does 3 mean?

Comment (tsunyoku) [reply app/handlers/profiles/settings/profbackground.go:175]: fuck if i know lol. this is old code

File: app/handlers/profiles/profile.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ func UserProfilePageHandler(c *gin.Context) {
 		services.DB.Get(&profileBackground, "SELECT type, value FROM profile_backgrounds WHERE uid = ?", data.UserID)
 		switch profileBackground.Type {
 		case 1, 3:
-			data.BannerContent = "/static/images/profbackgrounds/" + profileBackground.Value
+			data.BannerContent = "https://a.akatsuki.gg/profile-backgrounds/" + profileBackground.Value
```
Comment (cmyui) [inline app/handlers/profiles/profile.go]: do we have an env var for the base url? i know there are a couple of these

Comment (tsunyoku) [reply app/handlers/profiles/profile.go]: we do indeed 😭 let me fix

File: app/handlers/profiles/settings/profbackground.go [hunk 3]
```diff
@@ -74,7 +81,29 @@ func ProfileBackgroundSubmitHandler(c *gin.Context) {
 			slog.ErrorContext(c, err.Error())
 			return
 		}
-		saveProfileBackground(ctx, 1, fmt.Sprintf("%d.jpg?%d", ctx.User.ID, time.Now().Unix()))
```
Comment (cmyui) [inline app/handlers/profiles/settings/profbackground.go:77]: why was the unix timestamp attached to the end? is that important?

Comment (tsunyoku) [reply app/handlers/profiles/settings/profbackground.go:77]: ripple attempts to bypass caching pt 1 of 90

File: app/handlers/profiles/settings/profbackground.go [hunk 4]
```diff
@@ -74,7 +87,29 @@ func ProfileBackgroundSubmitHandler(c *gin.Context) {
...
+		f.Seek(0, io.SeekStart)
+
+		sess := session.Must(session.NewSession(&aws.Config{
+			Region:   aws.String(settings.AWS_REGION),
+			Endpoint: aws.String(settings.AWS_ENDPOINT_URL),
+		}))
+		uploader := s3manager.NewUploader(sess)
+		_, err = uploader.Upload(&s3manager.UploadInput{
+			Bucket:      aws.String(settings.AWS_BUCKET_NAME),
+			Key:         aws.String(fmt.Sprintf("profile-backgrounds/%d.jpg", ctx.User.ID)),
```
Comment (cmyui) [inline app/handlers/profiles/settings/profbackground.go:100]: i think u will need to create the directory in wasabi ui

Comment (tsunyoku) [reply app/handlers/profiles/settings/profbackground.go:100]: i don't think that's how it works?  but not confident.. can u do it, i'm on mobile

**Review by cmyui — APPROVED** [review]

File: app/handlers/profiles/settings/profbackground.go (hunk 4 above)
Comment (cmyui) [reply app/handlers/profiles/settings/profbackground.go:100]: yup will do it just in case

also we'll need to backfill here - i can support

Comment (tsunyoku) [reply app/handlers/profiles/settings/profbackground.go:100]: yep we can do that at some point

Comment (cmyui) [reply app/handlers/profiles/settings/profbackground.go:100]: directory creation done

File: app/handlers/profiles/settings/profbackground.go (hunk 3 above)
Comment (cmyui) [reply app/handlers/profiles/settings/profbackground.go:77]: kk we should test this but not a blocker/can do once the initial fix is out

Comment (cmyui) [reply app/handlers/profiles/settings/profbackground.go:77]: i think overall for `a.akatsuki.pw` we might want to send `cache-control: no-cache` headers

**Pull request merged**